controller/scheduler/swarm.py

```diff
@@ -1,21 +1,22 @@
 import re
 import time
-
+import logging
 from django.conf import settings
+from docker import errors
 from docker import Client
-
+from api.models import log_event
 from .states import JobState
 
-
 MATCH = re.compile(
     r'(?P<app>[a-z0-9-]+)_?(?P<version>v[0-9]+)?\.?(?P<c_type>[a-z-_]+)?.(?P<c_num>[0-9]+)')
 
 
 class SwarmClient(object):
 
-    def __init__(self, target, auth, options, pkey):
+    def __init__(self, target, auth, options, pkey, app):
```

Comment:
Catching API error in .create() is somewhat difficult as we call it in different threads. But if we could catch the api Error in .create() in models.py then passing the same error to the client is easy, which was actually discussed in #3605. This is a temporary fix just to make sure somehow users understand the reason for roll back